malformed/test_correct_headers.py

```diff
@@ -0,0 +1,181 @@
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80'
+        },
+    ]
+
+    def inner_test_correct_headers(self, chunksize, request):
+        # This function defines a test scenario itself.
+        # @chunksize and @request are required positional
+        # parameters
+        #
+        # In this partucular example the test reproduses
+        # the fills_hdr_tbl_for_req() test from unit tests.
+        #
+        # In your case there could be your own scenario.
+        # And it can have additional positional or keyword
+        # parameters
+        deproxy_srv = self.get_server('deproxy')
+        deproxy_srv.start()
+        self.start_tempesta()
+        deproxy_cl = self.get_client('deproxy')
+        deproxy_cl.segment_size = chunksize
+        deproxy_cl.start()
+        self.deproxy_manager.start()
+        self.assertTrue(deproxy_srv.wait_for_connections(timeout=1))
+        deproxy_cl.make_request(request)
+        self.assertTrue(deproxy_cl.valid_req_num != 0,
+                "Request was not parsed by deproxy client")
+        has_resp = deproxy_cl.wait_for_response(timeout=5)
+        self.assertTrue(has_resp, "Response not received"
+                        + "; with chunk size = " + str(chunksize))
+        status = int(deproxy_cl.last_response.status)
+        self.assertTrue(status == 200, "Wrong status: " + str(status)
+                                        +  ", expected: 200"
+                        + "; with chunk size = " + str(chunksize))
+        self.assertFalse(deproxy_srv.last_request is None,
+                           "Request was not send to backend"
+                        + "; with chunk size = " + str(chunksize))
+	req = deproxy.Request(request)
+	hdrs = req.headers.iteritems()
+        req2 = deproxy_srv.last_request
+        hdrs2 = req2.headers
+        for hdr in hdrs:
+            v2 = hdrs2.get(hdr[0], "-")
+            self.assertTrue(hdr[1] == v2, 
+                        "Header " + hdr[0] + " mismatch ("
+                                          + v2 + " != " + hdr[1] + ")"
+                        + "; with chunk size = " + str(chunksize))
+
+    def iterate_test(self, test_func, request, *args, **kwargs):
+        # This function provides iterations over vrious chunk sizes.
+        # It hase required positional parameters:
+        # @test_func - a function to iterate. Supposed, it implements
+        #              a test scenario and is a member of the same class.
+        # @request - a request to send.
+        # The function can accept additional positional and keyword
+        # parameters with *args and **kwargs to forward them to
+        # the @test_func().
+        CHUNK_SIZES = [ 1, 2, 3, 4, 8, 16, 32, 64, 128, 256, 1500, 9216, 1024*1024 ]
+        for i in range(len(CHUNK_SIZES)):
+            test_func(CHUNK_SIZES[i], request, *args, **kwargs)
+            if CHUNK_SIZES[i] > len(request):
+                break;
```

Comment:
Why do we need the test if we have just the same unit test, which works much faster? #115 is about a new transport option for deproxy, just like we have TLS and plain text. Tests like this one are possible, but it's not the main scenario which is supposed for functional test. More likely scenario is https://github.com/tempesta-tech/tempesta/issues/810#issuecomment-966690711 : just switch on chunked mode, send a chunked PURGE request and just verify that a cached response were purged. We do not care what was received - it Tempesta FW normally ate the chunked request a did what expected, then the test is passed. It is expected that a lot of tests will just use the transport along with TLS.

To make the life easier for other people, I just want to make a one new flag for deproxy server or client, that the peer must chunk all the data, and pass a generator function for the chunks sequence, most likely just returning 1.

With chunk sized I didn't mean that all the tests must try many variants of different chunking (e.g. one test with all the chunks of size 1, the next test for chunks of size 2 etc.). A developer may copy & paste tests with different generators it they want. I meant that generator must return the size for each chunk in the test. E.g. a developer may set a generator with random chunk size and each chunk will be of different random size.

`MalformedCrlfTest.common_check` is perfect: we set just one flag and we have a chunked version. But I'd like to be able to write something like
```Python
deproxy_cl.segment_size = random_numbers_generator
```

Reply:
This comment is not addressed